### PR TITLE
Update food-and-agriculture-organization-of-the-united-nations.csl

### PR DESCRIPTION
**Pull request by swifterslb**
CSL file for FAO House Style.

### CODE REVIEW

File: fao-house-style.csl [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  </locale>
+  <macro name="editor">
+    <names variable="editor" delimiter="," suffix=".">
+      <name and="symbol" et-al-min="4" initialize-with="."/>
+      <label form="short" strip-periods="false" prefix=", "/>
+    </names>
+  </macro>
+  <macro name="author">
+    <names variable="author" suffix=". ">
+      <name and="symbol" delimiter-precedes-et-al="never" delimiter-precedes-last="never" et-al-min="19" et-al-use-first="19" initialize-with="." name-as-sort-order="all"/>
```
Comment (adam3smith) [inline fao-house-style.csl]: could you set the et al mainly in cs:bibliography and cs:citation (where they're already included, you're just overriding them here). This is very hard to follow. You can leave the one in editor above if it's correct that you want et al for 4+ editors but only for 19+ authors in the bibliography.

File: fao-house-style.csl [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      </substitute>
+    </names>
+  </macro>
+  <macro name="author-short">
+    <names variable="author">
+      <name form="short" and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never" et-al-min="4" initialize-with="."/>
+      <et-al font-style="italic" prefix=" "/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
```
Comment (adam3smith) [inline fao-house-style.csl]: you almost certainly want to either remove this here or add it between lines 42 and 43 for the author-short macro.

File: fao-house-style.csl [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      <text variable="publisher-place" suffix=","/>
+      <text variable="publisher" prefix=" " suffix="."/>
+    </group>
+  </macro>
+  <macro name="edition">
+    <number font-style="italic" vertical-align="baseline" suffix=" " variable="edition" form="ordinal"/>
+    <text term="edition" font-style="italic" suffix=". "/>
+  </macro>
+  <macro name="Available-at-and-accessed">
+    <text term="available at" text-case="capitalize-first" vertical-align="baseline" suffix=": "/>
```
Comment (adam3smith) [inline fao-house-style.csl]: remove the vertical-align here and in line 65. It doesn't actually do anything.

File: fao-house-style.csl [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        </if>
+        <else>
+          <text term="no date" form="short"/>
+        </else>
+      </choose>
+      <choose>
+        <if type="bill book graphic legal_case legislation motion_picture song" match="any">
+          <choose>
+            <if match="any" is-numeric="edition">
+              <group delimiter=" " prefix=" ">
```
Comment (adam3smith) [inline fao-house-style.csl]: way too complex here: Why not just wrap title and edition into a group with delimiter=", " and suffix=". " and remove the suffix=". " from the edition term above.

File: fao-house-style.csl [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        <else-if type="thesis" match="any">
+          <text variable="title" font-style="italic" suffix="."/>
+          <text macro="publisher"/>
+          <text variable="genre" prefix=" (" suffix=")"/>
+        </else-if>
+        <else>
+          <text variable="title" prefix=" " suffix="."/>
+          <group prefix=" ">
+            <text variable="container-title" form="short" font-style="italic" suffix=", "/>
+            <text variable="volume" form="short" font-weight="normal"/>
```
Comment (adam3smith) [inline fao-house-style.csl]: remove the font-weights here, and next line. They're not doing anything.

**Comment by adam3smith** [comment]
More generally and for naming: What exactly is this for and approx. how many people will be using it? We'll probably just want it named "Food and Agriculture Organization of the United Nations" but I'd like to know a little more about this. We're also reluctant to accept limited circulation in-house styles, but normally at the size of a UN organization that should be fine (we do host the CEPAL/ECLAC style).

**Comment by adam3smith** [comment]
You can update the style by using the pencil icon at https://github.com/swifterslb/styles/tree/patch-1/fao-house-style.csl
Let us know when you're done or if you have any questions&also see my questions above.

**Comment by swifterslb** [comment]
Thank you so much for your kind help! I have no programming/scripting experience at all except some very simple HTML. I just used the Visual CSL editor mostly and didn't really check for redundancy, bugs, etc. GitHub is so great a place!

I've made all the changes you pointed out, except that I decided to include et-al settings in author and author-short macros and remove them from citation and bibliography fields.

Now for usage. This is more likely a personal project. I work at FAO and I've recently been writing for a publication. I'm surprised to find that there is no official (or at least publicly available) style file for either EndNote or Zotero. I asked senior colleagues and they either didn't care (leaving all the styling job to the editors) or didn't know. Requesting the publication people to develop one would be fruitless almost for sure, and I always hate EndNote, so I decided to write one for Zotero myself. Each year FAO produces so many publications (http://www.fao.org/publications/en/) and all the papers, reports, and books among them need to follow the style. Therefore, I believe it is of some importance.

The thing is, Zotero hasn't been widely adopted here. I've seen some senior colleagues just leaving all the citations blank or in some other awkward way. This is just unbelievable in 2015. I hope to promote the use of Zotero in my organization not only because it's free, but also because it has a standalone, installation-free version working perfectly (FAO staffs don't have administrative privileges on working computers for security reasons). This style file is an integral part of the solution.

For the name, I use "FAO House Style" just because the guidelines (http://www.fao.org/docrep/004/ac339e/AC339E01.htm#P207_14907) say so. If you feel uncomfortable with "house" of course I can change it.

Thanks again and nice to meet you.

Regards,
Bin

On 23/08/2015 00:51:43, Sebastian Karcher <notifications@github.com> wrote:
You can update the style by using the pencil icon at https://github.com/swifterslb/styles/tree/patch-1/fao-house-style.csl [https://github.com/swifterslb/styles/tree/patch-1/fao-house-style.csl]
Let us know when you're done or if you have any questions&also see my questions above.

**Comment by adam3smith** [comment]
Great, thanks.
@rmzelle -- what do you think about naming? 
I think potential usage is definitely large enough to warrant inclusion, yes.

**Comment by rmzelle** [comment]
I would prefer:

```xml
<title>Food and Agriculture Organization of the United Nations</title>
<title-short>FAO</title-short>
```

and "food-and-agriculture-organization-of-the-united-nations" as file name/style ID/"self" link.

**Comment by swifterslb** [comment]
Thanks. I'll change that. I'm still updating it for some other article formats but encounter a problem: How can I show a date range, e.g. "26–28 June 2002"? CSL guide has it (http://docs.citationstyles.org/en/stable/specification.html#date-ranges) but how should I input the date in Zotero? I've tried many ways without success. Has this function been made available yet?

**Comment by adam3smith** [comment]
no, date ranges aren't currently supported in Zotero.

**Comment by swifterslb** [comment]
Updated the names. Updated settings for several article types. Please check.

**Comment by rmzelle** [comment]
Looks good from my end.

File: food-and-agriculture-organization-of-the-united-nations.csl [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+        </else-if>
+        <else-if type="article-newspaper article-magazine" match="any">
+          <text variable="title" suffix=". "/>
+          <text variable="container-title" form="short" font-style="italic" suffix="."/>
+          <group prefix=" ">
+            <choose>
+              <if match="any" variable="URL">
+                <text macro="Available-at-and-accessed" prefix=" "/>
+              </if>
+              <else>
```
Comment (adam3smith) [inline food-and-agriculture-organization-of-the-united-nations.csl]: this doesn't do anything, you can just delete lines 163-165.

File: food-and-agriculture-organization-of-the-united-nations.csl [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        </else-if>
+        <else-if type="report" match="any">
+          <text variable="title" font-style="italic" suffix="."/>
+          <text macro="publisher"/>
+          <group prefix=" ">
+            <choose>
+              <if match="any" variable="URL">
+                <text macro="Available-at-and-accessed"/>
+              </if>
+              <else>
```
Comment (adam3smith) [inline food-and-agriculture-organization-of-the-united-nations.csl]: same here for 177-179: an "else don't print anything" is always implied in if/else-if elements.

File: food-and-agriculture-organization-of-the-united-nations.csl [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+          <group delimiter=", " prefix=" " suffix=".">
+            <text variable="event"/>
+            <text variable="event-place"/>
+            <date form="text" date-parts="year-month-day" variable="issued"/>
+          </group>
+          <choose>
+            <if match="any" variable="URL">
+              <text macro="Available-at-and-accessed" prefix=" "/>
+            </if>
+            <else>
```
Comment (adam3smith) [inline food-and-agriculture-organization-of-the-united-nations.csl]: ditto

**Comment by adam3smith** [comment]
please remove those extra else elements you've added, otherwise good by me, too.

**Comment by swifterslb** [comment]
Thanks for the help! I thought if you have an "if", an "else" action is mandatory... They're deleted now.

**Comment by adam3smith** [comment]
thanks!

**Comment by rmzelle** [comment]
:sparkles:
